Sources/DistributedActors/ClusterSystem.swift

```diff
@@ -1019,20 +1029,98 @@ public struct ClusterInvocationResultHandler: DistributedTargetInvocationResultH
     }
 
     public func onReturn<Success: Codable>(value: Success) async throws {
-        let ref = _ActorRef<Success>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
-        ref.tell(value)
+        let ref = _ActorRef<RemoteCallReply<Success>>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
+        ref.tell(.init(value: value))
     }
 
     public func onReturnVoid() async throws {
-        let ref = _ActorRef<_Done>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
-        ref.tell(_Done.done)
+        let ref = _ActorRef<RemoteCallReply<_Done>>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
+        ref.tell(.init(value: _Done.done))
     }
 
     public func onThrow<Err: Error>(error: Err) async throws {
         self.system.log.warning("Result handler, onThrow: \(error)")
-        // FIXME(distributed): carry the error back
-        fatalError("FIXME: implement sending back errors")
+        let ref = _ActorRef<RemoteCallReply<_Done>>(.remote(.init(shell: clusterShell, address: replyToAddress, system: system)))
+        if let codableError = error as? (Error & Codable) {
+            ref.tell(.init(error: codableError))
+        } else {
+            ref.tell(.init(error: GenericRemoteCallError(message: "Remote call error of [\(type(of: error as Any))] type occurred")))
+        }
+    }
+}
+
+protocol AnyRemoteCallReply: Codable {
+    associatedtype Value: Codable
+
+    var value: Value? { get }
+    var thrownError: (any Error & Codable)? { get }
+
+    init(value: Value)
+    init<Err: Error & Codable>(error: Err)
```

Comment:
do we need the inits in the protocol? I'd expect not to need them, and only need the getters 🤔

Reply:
It's used in [`_tellOrDeadLetter`](https://github.com/apple/swift-distributed-actors/pull/931/files#diff-19182eaca69e6caa9693cf4f221489ea4d4d1cde2644dad95726066984189aaaR247) to construct new `RemoteCallReply`. Open to suggestions.

Reply:
Oh I see! Super sneaky, I love it -- good to keep 👍